### ISSUE TITLE
Incompatibility between react-pdf `PDFDocumentProxy` and pdfjs-dist `PDFDocumentProxy`

### ISSUE THREAD

**Issue by fjwillemsen** [wojtekmaj/react-pdf#1801]
### Before you start - checklist

- [X] I followed instructions in documentation written for my React-PDF version
- [X] I have checked if this bug is not already reported
- [X] I have checked if an issue is not listed in [Known issues](https://github.com/wojtekmaj/react-pdf/wiki/Known-issues)
- [ ] If I have a problem with PDF rendering, I checked if my PDF renders properly in [PDF.js demo](https://mozilla.github.io/pdf.js/web/viewer.html)

### Description

It appears that there is a discrepancy between the `PDFDocumentProxy` defined by react-pdf and the `PDFDocumentProxy` defined by pdfjs-dist (4.2.67). This causes a type incompatibility between the two. 

### Steps to reproduce

Minimal example:

```
import { Document, Page } from "react-pdf";
import type { PDFDocumentProxy } from "pdfjs-dist/types/src/display/api";

const handleDocumentLoadSuccess = useCallback((pdf: PDFDocumentProxy) => {
    console.log(pdf);
}, []);

<Document file={pdf.url || "/blank.pdf"} onLoadSuccess={handleDocumentLoadSuccess} loading="">
    <Page
        renderAnnotationLayer={true}
        renderTextLayer={false}
        loading=""
    />
</Document>
```

### Expected behavior

After loading the PDF, I expect to see a console log statement.

### Actual behavior

The following error is produced:
```
Type '(pdf: PDFDocumentProxy) => void' is not assignable to type 'OnDocumentLoadSuccess'.
  Types of parameters 'pdf' and 'document' are incompatible.
    Property '<user>/node_modules/react-pdf/node_modules/pdfjs-dist/types/src/display/api").PDFDocumentProxy' but required in type 'import("<user>/node_modules/pdfjs-dist/types/src/display/api").PDFDocumentProxy'.
```

In addition, if `renderAnnotationLayer` is set to `false`, the following error is produced:
```
Type '(pdf: PDFDocumentProxy) => void' is not assignable to type 'OnDocumentLoadSuccess'.
  Types of parameters 'pdf' and 'document' are incompatible.
    Type 'import("<user>/node_modules/react-pdf/node_modules/pdfjs-dist/types/src/display/api").PDFDocumentProxy' is not assignable to type 'import("<user>/node_modules/pdfjs-dist/types/src/display/api").PDFDocumentProxy'.
      Types of property 'annotationStorage' are incompatible.
        Type 'AnnotationStorage' is missing the following properties from type 'AnnotationStorage': editorStats, #private
```

### Environment

- **Browser (if applicable)**: N/A
- **React-PDF version**: 8.0.2
- **React version**: 18.3.1
- **Webpack version (if applicable)**: N/A

**Comment by wojtekmaj** [comment]
No it's not. pdfjs-dist is an indirect dependency via React-PDF. I guess it would be better if we re-exported it... 🤔

**Comment by fjwillemsen** [comment]
Yes, it seems that the dependency [`pdfjs-dist` is fixed to "3.11.174",](https://github.com/wojtekmaj/react-pdf/blob/81c98f633f104fd28a0d9537a48e9d1b28cc9ea1/packages/react-pdf/package.json#L63) whereas [`pdfjs-dist` has been on >4.0 since 7 months](https://www.npmjs.com/package/pdfjs-dist?activeTab=versions), causing the discrepancy.

**Comment by wojtekmaj** [comment]
No, the discrepancy is caused by installing an incompatible version of pdfjs-dist manually. React-PDF must use a fixed version because to Mozilla, these version numbers are meaningless and there are breaking changes between minor versions.
